test/terraform/gke_test.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package terraform
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var project string
+var tfPass string
+
+func TestTerraformGKEInstallConfig(t *testing.T) {
+	t.Logf("Using project %s\n", project)
+	_, err := os.Stat(tfPass)
+	assert.NoError(t, err)
+
+	terraformOptions := &terraform.Options{
+		TerraformDir: tfPass,
+		Vars: map[string]interface{}{
+			"project":     project,
+			"name":        "terratest-cluster2",
+			"values_file": "",
+		},
+	}
+	// run 'terraform apply' command
+	// this would create GKE cluster,
+	// deploy Agones of a release version and
+	// create kubeconfig file
+	terraform.InitAndApply(t, terraformOptions)
+
+	defer func() {
+		err := destroy(t, terraformOptions, tfPass)
+		if err != nil {
+			assert.FailNow(t, err.Error())
+		}
+	}()
+
+	output := terraform.Output(t, terraformOptions, "host")
+	assert.Contains(t, output, "https://")
+
+}
+
+// destroy runs terraform destroy on agones helm deployment.
+// Second step is to run terraform destroy for all other resources including
+// pregenerated kubeconfig file
+func destroy(t *testing.T, options *terraform.Options, tfPass string) error {
+	options.Targets = []string{"module.gke_helm.module.helm_agones.helm_release.agones"}
```

Comment:
I take it we're still seeing the same issue with Helm 3? Such that we have to delete Helm first, otherwise we leave behind orphaned loadbalancers?

Reply:
Doing this test with 
```
func destroy(t *testing.T, options *terraform.Options, tfPass string) error {
	options.Targets = []string{}
	terraform.Destroy(t, options)
	return nil
}
```

Reply:
Yes, now it is working in just `terraform destroy`.